hook!-full-game/src/Player/CameraRig.gd

```diff
@@ -2,18 +2,35 @@ extends Position2D
 
 
 onready var _camera: Camera2D = $Camera2D
+onready var _tween : Tween = $Tween
 
 export var offset: = Vector2(200.0, 160.0)
 export var mouse_range: = Vector2(100.0, 500.0)
 
+var inactive := false
 
 func _ready() -> void:
 	assert mouse_range.x < mouse_range.y
 	assert mouse_range.x >= 0.0 and mouse_range.y >= 0.0
+	Events.connect("player_died", self, "_on_Events_player_died")
+
+
+func _on_Events_player_died() -> void:
+	inactive = true
+	var screen_center := to_global(Vector2.ZERO)
+	_tween.interpolate_property(_camera, "zoom",
```

Comment:
Following on why you shouldn't have animation via code, you actually have the animation split in 2 places. With the AnimationPlayer, it's just 1 animation with all the data in 1 place.